### DISCUSSION TITLE
Why not release our own Math Expression Parser library ?

### DISCUSSION THREAD

**Discussion by clementwzk** [clementwzk/OpenCalc#434] [General]
OpenCalc has been using its own parser for several months. It is now becoming increasingly robust. It would be interesting to publish it as a library like mXparser, for example. Obviously the current version lacks a lot of features and it would require a lot of work. What do you think? Personally, I have no experience of creating a library.

**Comment by inson1** [comment]
Its good idea, but maybe little bit too early

New version of mXparser isnt FOSS https://github.com/Yet-Zio/yetCalc/issues/98, so it would good alternative for that

Also depends on your time, it would be more work

What about keeping it without Android specific code?

And with library org makes more and more sense https://github.com/Darkempire78/OpenCalc/issues/202

Also the library would become more bloated as more developers would neeed more features, so mor size for OpenCalc 

(also what about new release?)

**Comment by clementwzk** [comment]
(I will release the new version today)

**Comment by clementwzk** [comment]
> What about keeping it without Android specific code?

Sure, currently it does not require any android package if I'm not wrong. BTW idk if we can support as many languages than Mxparser does

> And with library org makes more and more sense https://github.com/Darkempire78/OpenCalc/issues/202

Yes
